File: 1260/main.cpp
#include <iostream>
#include <queue>

using namespace std;

const int MAX = 1001;

int mat[MAX][MAX];
int visit[MAX];
int N, M , V;

int main(void) {

	cin >> N >> M >> V;

	for ( int i = 0; i < M ; i++ ) {
		int first, second;
		cin >> first >> second;
		mat[first][second] = mat[second][first] = 1;
	}

}

void dfs (int v) {
	cout << v << " ";
	visit[v] = 1;
	for ( int i = 1; i <= N ; i++ ) {
		if ( visit[v] == 1 || mat[v][i] == 0 )
			continue;
		dfs(i);
	}
}	

void bfs (int v) {
	queue<int> q;
	visit[v] = 0;

}

File: 1260/main.py
from collections import deque

# read N, M, V
N, M, V = map(int, input().split())
mat = [[0 for col in range(N+1)] for row in range(N+1)]
visited = [0 for col in range(N+1)]

# read start, end nodes
for row in range(M):
    start, end = map(int, input().split())
    mat[start][end] = mat[end][start] = 1

# dfs
def dfs(v):
   visited[v] = 1
   print(v, end=' ')
   for idx in range(1, N+1):
       if mat[v][idx] == 1 and visited[idx] == 0:
           dfs(idx)

#BFS
def bfs(start_v):
    discovered = [start_v]

    queue = deque()
    queue.append(start_v)

    while queue:
        v = queue.popleft()
        print(v, end=' ')

        for w in range(1 , N+1):
            if mat[v][w] == 1 and (w not in discovered):
                discovered.append(w)
                queue.append(w)


dfs(V)
print()
bfs(V)


